wiring/inc/spark_wiring_ble.h

```diff
@@ -395,9 +402,13 @@ class BleCharacteristic {
     ssize_t setValue(const uint8_t* buf, size_t len);
     ssize_t setValue(const String& str);
     ssize_t setValue(const char* str);
+    ssize_t setValue(const uint8_t* buf, size_t len, bool ack);
+    ssize_t setValue(const String& str, bool ack);
+    ssize_t setValue(const char* str, bool ack);
 
     template<typename T>
-    ssize_t setValue(T val) {
+    typename std::enable_if<std::is_integral<T>::value, ssize_t>::type
```

Comment:
This also prevents things like:
```cpp
struct TestStruct {
    int data;
    bool data1;
};
TestStruct test;
XXX.setValue(test);
```

Is this something we want to do? Perhaps this should be `is_standard_layout` instead.

Reply:
I don't think we support sending this kind of struct for now, besides, it may send random data due to struct alignment.

Reply:
All I'm saying is that right now it's possible and whether we should introduce a restriction.

> it may send random data due to struct alignment.

Or it may not if both sides follow the format.

Reply:
If using `is_standard_layout`, it causes ambiguous calling to `setValue(const uint8_t* buf, size_t len)`. BTW, `setValue(const T& val)` would be better in comparison to `setValue(T val)`. Also type restriction should also apply to the `getValue(T* val)` template.

Reply:
> it causes ambiguous calling to

That's weird. It does handle things like `uint8_t[123]`, but there is no `size_t` argument. It might introduce ambiguity 
to `const char*` variant I would expect if passed string is `const char[]`.

I've commented on `getValue` in another PR. I think it's fine to leave things as is and require the users to cast to `uint8_t*`, however we might as well instead:

1. `std::is_standard_layout && !std::is_array` to perhaps allow standard layout structs and scalars and disallow arrays
2. Or potentially "simply" exclude ambiguous types (I would expect `uint8_t[]` and `char[]`) to also allow arrays of standard layout types. It's somewhat convoluted in C++14, but should be easier to do in C++17 with `if constexpr()`.

> setValue(const T& val) would be better

Perhaps, practically there shouldn't be any difference in optimizations.

I propose we simply go with the first approach for now in order to keep the existent behavior. Might as well make to `const T&`.

Reply:
The ambiguity I mentioned is introduced due to `setValue(const T& val, bool ack)`.